lib/SIL/SILVerifier.cpp

```diff
@@ -4302,6 +4302,32 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     }
   }
 
+  void verifyLoadInitialization(SILFunction *F) {
+    llvm::SmallSet<SILValue, 4> AllocInstSet;
+    for (auto &BB : *F) {
+      AllocInstSet.clear();
+      for (auto &I : BB) {
+        if (auto *SI = dyn_cast<StoreInst>(&I)) {
+          if (AllocInstSet.count(SI->getOperand(1)))
+            AllocInstSet.erase(SI->getOperand(1));
+        }
+        else if (auto *LI = dyn_cast<LoadInst>(&I)) {
+          require(!AllocInstSet.count(LI->getOperand()),
+                  "possibly uninitialized load");
+        }
+        else if (auto *ASI = dyn_cast<AllocStackInst>(&I)) {
+          AllocInstSet.insert(ASI);
+        }
+        else {
```

Comment:
Please add  a comment here that you conservatively assume that all uses of the alloc_stack either store to the location or let the address escape. Which means that the alloc_stack location is potentially initialized.